bytecomp/switch.ml

```diff
@@ -151,12 +154,26 @@ end
 *)
 module Make (Arg : S) =
 struct
+  type case = {
+    low_loc : Arg.location;
+    low : int;
+    high_plus_one_loc : Arg.location;
```

Comment:
I don't understand the purpose of `low_loc` and `high_plus_one_loc`. They add some complexity to `matching.ml`, and make `switch.ml` much harder to read; and the code dealing with the `plus_one` aspect of `high_plus_one_loc` often looks shady. They may be useful, but I don't understand what they mean and why we need them.

If I understand correctly, a `case` represents a contiguous sequence of integers on which we are switching, that are sent to various exits given by the `action` array. What are `low_loc` and `high_plus_one_loc` supposed to represent? Could they be replaced by simply fetching locations inside `action_index`? (or maybe a separate index for locations?)

Reply:
I think what we need here is someone who is an expert on this file, which I am definitely not, to help.  The locations aren't corresponding to the actions---they're corresponding to the constants being matched against (or at least, that was the intention).  I suggest we resolve `matching.ml` first and then find someone to help with this part.